File: CF_Quick-One/280_div2/C.py
from sys import stdin
def input(): return stdin.readline()[:-1]

N, MAX, AVG = map(int, input().split())
ans = 0
tot = AVG * N
exams = []
for _ in range(N):
    a, b = map(int, input().split())
    tot -= a
    exams.append((b, a))

if tot <= 0:
    print(0)
    exit()

exams.sort()
for b, a in exams:
    can_be_done = MAX - a
    freq = min(can_be_done, tot)
    ans += freq * b
    tot -= freq
print(ans)

File: CF_Quick-One/280_div2/C.cpp
#include <bits/stdc++.h>

using namespace std;

#define int long long

int32_t main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, r, avg;
    cin >> n >> r >> avg;
    vector<pair<int, int>> v(n);
    int ans = 0;
    int goal = avg * n;

    for (int i = 0; i < n; i++)
    {
        int a, b;
        cin >> a >> b;
        goal -= a;
        v.push_back({b, r - a});
    }

    if (goal <= 0){
        cout << 0 << endl;
        return 0;
    }

    sort(v.begin(), v.end());
    for (auto& p: v){
        int freq = min(goal, p.second);
        goal -= freq;
        ans += freq * p.first;
    } 

    cout << ans << endl;
    return 0;
}